allium/lib/search_discovery.py

```
@@ -125,12 +182,20 @@ def generate_search_discovery(output_dir, base_url):
+            "html_count": 0,
+            "noindex_count": 0,
+        }
 
     output_path.mkdir(parents=True, exist_ok=True)
-    urls = _generated_urls(output_path, public_base_url)
+    urls, html_count, noindex_count = _canonical_urls_from_html(
+        output_path, public_base_url
+    )
     if not urls:
-        raise ValueError("cannot generate a sitemap without public HTML routes")
+        raise ValueError("cannot generate a sitemap without indexable canonical URLs")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)